agora/Agora/Proposal/Scripts.hs

```diff
@@ -236,339 +239,433 @@ proposalValidator proposal =
+                      _ -> pconstant True
 
-          pguardC "Cannot advance ahead of time" notTooEarly
-          pguardC "Finished proposals cannot be advanced" $ pnot # isFinished
+                pguardC "Cannot advance ahead of time" notTooEarly
+                pguardC "Finished proposals cannot be advanced" $ pnot # isFinished
 
-          thresholdsF <- pletFieldsC @'["execute"] proposalF.thresholds
+                thresholdsF <- pletFieldsC @'["execute", "vote"] proposalF.thresholds
 
-          pure $
-            pif
-              notTooLate
-              -- On time: advance to next status.
-              ( pmatchEnum proposalStatus $ \case
-                  Draft -> unTermCont $ do
-                    -- TODO: Perform other necessary checks.
-
-                    -- 'Draft' -> 'VotingReady'
-                    pguardC "Proposal status set to VotingReady" $
-                      proposalOutStatus #== pconstantData VotingReady
-
-                    pure $ popaque (pconstant ())
-                  VotingReady -> unTermCont $ do
-                    -- 'VotingReady' -> 'Locked'
-                    pguardC "Proposal status set to Locked" $
-                      proposalOutStatus #== pconstantData Locked
-
-                    pguardC "Winner outcome not found" $
-                      pisJust #$ pwinner' # proposalF.votes
-                        #$ punsafeCoerce
-                        $ pfromData thresholdsF.execute
-
-                    pure $ popaque (pconstant ())
-                  Locked -> unTermCont $ do
-                    -- 'Locked' -> 'Finished'
-                    pguardC "Proposal status set to Finished" $
-                      proposalOutStatus #== pconstantData Finished
-
-                    -- TODO: Perform other necessary checks.
-                    pure $ popaque (pconstant ())
-                  _ -> popaque (pconstant ())
-              )
-              -- Too late: failed proposal, status set to 'Finished'.
-              ( popaque $
-                  ptraceIfFalse "Proposal should fail: not on time" $
-                    proposalOutStatus #== pconstantData Finished
-                    -- TODO: Should check that the GST is not moved
-                    --        if the proposal is in 'Locked' state.
-              )
+                pure $
+                  pif
+                    notTooLate
+                    -- On time: advance to next status.
+                    ( pmatchEnum proposalStatus $ \case
+                        Draft -> unTermCont $ do
+                          -- 'Draft' -> 'VotingReady'
+                          pguardC "Proposal status set to VotingReady" $
+                            proposalOutStatus #== pconstantData VotingReady
+
+                          pguardC "Should exceed minimum cosign amount" $
+                            pfromData thresholdsF.vote
+                              #< pfromData proposalF.cosignAmount
```

Comment:
Here, then instead of checking the amount, we'd just check the `Stake` datums for their total GT.